test/gtest/fusion_test.cpp

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+
+#if MIOPEN_BACKEND_HIP // equiv to hipInit(0)?
+
+std::vector<int> strides(std::vector<int> dims)
+{
+    std::vector<int> strides_array(4);
+    int d = 1;
+    for(int i = 3; i >= 0; --i)
+    {
+        strides_array[i] = d;
+        d *= dims[i];
+    }
```

Comment:
This can be converted to a `std::transform` and is likely already implemented in one of the ctors for the `TensorDescriptor`

Reply:
It is a separate function https://github.com/ROCmSoftwarePlatform/MIOpen/blob/68524a8223a6550115589e2bd671d378d92fe470/src/tensor.cpp#L290 but how could I use it in construction initializer since the descriptor is a member inside tensor https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/test/tensor_holder.hpp#L111 ?